### PR TITLE
Feat: Ember Octane Update + Additional Features

### LINKED ISSUE #3: Handle dynamically introduced higher priority elements
If a higher priority sequential render instance is dynamically added to the template at a later point in time, give higher precedence to those before executing the elements in the current queue.

Say we have the following tasks.

0  => ['high_priority_render_1', 'high_priority_render_2']
1  => ['medium_priority_render_1',  'medium_priority_render_2']
2  => ['low_priority_render_1']

And when app is currently executing medium_task1 and another task 'high_priority_render_3' is added to the 0 queue, high_priority_render_3 should be picked up automatically by virtue of its priority.

### LINKED ISSUE #7: Ember octane update 🤓
1. Update the addon to octane
2. Remove the component lifecycle hook for better glimmer compatibility

### LINKED ISSUE #32: Introduce retry API instead of renderImmediately and triggerOutOfOrder
1. renderImmediately -> This argument immediately renders the content without triggering a data refetch or affecting the render queue
2. triggerOutOfOrder -> This triggers data fetch + render without modifying the app's render state or affecting the queue

Both these implementations are actions that are represented by arguments. This is not ideal.

So lets yield a retry action, that can easily replace these two functionalities and also have other powerful use cases. This action if triggered will not affect the current renderState or the renderQueues.

### PR DESCRIPTION
**Pull request by kanalveli-ramachandran**
Summary

1. Remove deprecated APIs - #10 
2. Fixes #7 
3. Glimmer + Native Class Migration
3. Fixes #3  
4. Fixes #32
5. ember-cli-addon-docs update

TODO

- [x] Add more strenuous tests
- [x] Documentation Refresher
- [x] Fix deprecation warnings

### CODE REVIEW

File: .template-lintrc.js [hunk 1]
```diff
@@ -6,9 +6,7 @@ module.exports = {
 
   rules: {
     'no-bare-strings': true,
-    'quotes': 'double'
+    quotes: 'double',
```
Comment (rajasegar-c) [inline .template-lintrc.js:9]: Usually we prefer single quotes right? Is Ember enforcing double quotes? any reason why?

File: config/ember-try.js [hunk 2]
```diff
@@ -1,81 +1,84 @@
...
+          },
+        },
       },
-      // The default `.travis.yml` runs this scenario via `npm test`,
-      // not via `ember try`. It's still included here so that running
-      // `ember try:each` manually or from a customized CI config will run it
-      // along with all the other scenarios.
       {
-        name: 'ember-default',
+        name: 'ember-default-with-jquery',
```
Comment (rajasegar-c) [inline config/ember-try.js:50]: do we still need jquery and ember-classic environments?

File: package.json [hunk 3]
```diff
@@ -38,46 +41,57 @@
...
-    "ember-cli-uglify": "^3.0.0",
+    "ember-cli-terser": "^4.0.2",
     "ember-disable-prototype-extensions": "^1.1.3",
-    "ember-export-application-global": "^2.0.0",
-    "ember-load-initializers": "^2.1.0",
+    "ember-export-application-global": "^2.0.1",
+    "ember-load-initializers": "^2.1.2",
     "ember-maybe-import-regenerator": "^0.1.6",
-    "ember-qunit": "^4.5.1",
+    "ember-page-title": "^6.2.2",
```
Comment (rajasegar-c) [inline package.json:73]: do we need ember-page-title?

**Review by rajasegar-c — CHANGES_REQUESTED** [review]
Please check the review comments

File: .template-lintrc.js (hunk 1 above)
Comment (kanalveli-ramachandran) [reply .template-lintrc.js:9]: In templates, the recommended convention for quotes is double ji. Only in js/ts we default to single quotes.

**Review by rajasegar-c — APPROVED** [review]

**Pull request merged**